cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,190 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Harry Potter and Philosopher's Stone",
+            "description": "Fantasy novel written by British author J. K. Rowling",
+            "duration": 120,
+        }
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "admin@gmail.com",
+            "admin_secret_password",
+            is_staff=True
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_create_movie(self):
+        payload = {
+            "title": "Harry Potter and Chamber of Secret",
+            "description": "Fantasy novel part 2, written by British author J. K. Rowling",
+            "duration": 110,
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=res.data["id"])
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        for key in payload:
+            self.assertEqual(payload[key], getattr(movie, key))
+
+    def test_create_movie_with_genres(self):
+        drama = sample_genre(name="Drama")
+        horror = sample_genre(name="Horror")
+
+        payload = {
+            "title": "Mirror",
+            "description": "Really scary movie",
+            "duration": 100,
+            "genres": [drama.id, horror.id]
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=res.data["id"])
+        genres = movie.genres.all()
+
+        self.assertEqual(genres.count(), 2)
+        self.assertIn(drama, genres)
+        self.assertIn(horror, genres)
+
+    def test_create_movie_with_actors(self):
+        actor_1 = sample_actor(first_name="Diana", last_name="Milli")
+        actor_2 = sample_actor(first_name="Symon", last_name="Rick")
+        actor_3 = sample_actor(first_name="Nick", last_name="Hophman")
+
+        payload = {
+            "title": "Imagination",
+            "description": "Ready for watching",
+            "duration": 150,
+            "actors": [actor_1.id, actor_2.id, actor_3.id]
+        }
+        res = self.client.post(MOVIE_URL, payload)
+        movie = Movie.objects.get(id=res.data["id"])
+        actors = movie.actors.all()
+
+        self.assertEqual(actors.count(), 3)
+        self.assertIn(actor_1, actors)
+        self.assertIn(actor_2, actors)
+        self.assertIn(actor_3, actors)
```

Comment:
this can be done in one `test_create_movie_with_actors_and_genres` or something like that